src/platform/user/profile/vet360/actions/transactions.js

```diff
@@ -164,10 +165,21 @@ export function createTransaction(
         ? await apiRequest(route, options)
         : await localVet360.createTransaction();
 
-      recordEvent({
-        event: method === 'DELETE' ? 'profile-deleted' : 'profile-transaction',
-        'profile-section': analyticsSectionName,
-      });
+      // We want the validateAddreses method handling dataLayer events for saving / updating addresses.
+      if (vaProfileUseAddressValidation(state)) {
+        if (!fieldName.toLowerCase().includes('address')) {
+          recordEvent({
+            event:
+              method === 'DELETE' ? 'profile-deleted' : 'profile-transaction',
+            'profile-section': analyticsSectionName,
+          });
+        }
+        recordEvent({
```

Comment:
doesn't this need to live in an `else` block when `vaProfileUseAddressValidation(state)` is `false`?

Reply:
^^ same question but I think you meant for the conditional `!fieldName.toLowerCase().includes('address')`, otherwise if that evaluates to true we will record 2 events (maybe that's desirable but just want to confirm!)